pkg/importer/vddk-datasource_amd64.go

```diff
@@ -855,35 +855,47 @@ func createVddkDataSource(endpoint string, accessKey string, secKey string, thum
 	// then get the list of changed blocks from VMware for a delta copy.
 	var changed *types.DiskChangeInfo
 	if currentSnapshot != nil && previousCheckpoint != "" {
-		// Check if this is a snapshot or a change ID, and query disk areas as appropriate.
-		// Change IDs look like: 52 de c0 d9 b9 43 9d 10-61 d5 4c 1b e9 7b 65 63/81
-		changeIDPattern := `([0-9a-fA-F]{2}\s?)*-([0-9a-fA-F]{2}\s?)*\/([0-9a-fA-F]*)`
-		if matched, _ := regexp.MatchString(changeIDPattern, previousCheckpoint); matched {
-			request := types.QueryChangedDiskAreas{
-				ChangeId:    previousCheckpoint,
-				DeviceKey:   backingFileObject.Key,
-				Snapshot:    currentSnapshot,
-				StartOffset: 0,
-				This:        vmware.vm.Reference(),
-			}
-			response, err := QueryChangedDiskAreas(vmware.context, vmware.vm.Client(), &request)
-			if err != nil {
-				return nil, err
-			}
-			changed = &response.Returnval
-		} else { // Previous checkpoint is a snapshot
-			previousSnapshot, err := vmware.vm.FindSnapshot(vmware.context, previousCheckpoint)
-			if err != nil {
-				klog.Errorf("Could not find previous snapshot %s: %v", previousCheckpoint, err)
-				return nil, err
-			}
-			if previousSnapshot != nil {
-				changedAreas, err := vmware.vm.QueryChangedDiskAreas(vmware.context, previousSnapshot, currentSnapshot, backingFileObject, 0)
+		// The QueryChangedDiskAreas returns maximally 2000 changed blocks. If the disk has more than 2000 blocks
+		// we need to query the next chunk of the blocks starting from previous
+		const MaximumQueryResponse = 2000
+		var startOffset int64 = 0
+		var lastChangedSize = 0
+		changed = &types.DiskChangeInfo{}
+		for changed.Length == 0 || lastChangedSize == MaximumQueryResponse {
```

Comment:
Can you make this two smaller loops instead? Like just around the actual QueryChangedDiskAreas calls. That will clean up one of the lint errors about MatchString and avoid repeated calls to FindSnapshot, which can be expensive.